Charlesmchen/voice messages ui

charlesmchen opened the pull request:
See: https://github.com/WhisperSystems/SignalServiceKit/pull/212

PTAL @michaelkirk

michaelkirk commented on Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m:
```diff
@@ -259,10 +204,149 @@ - (UIView *)mediaView {
+{
+    OWSAssert([self isAudio]);
+
+    return (self.attachment.isVoiceMessage || self.attachment.filename.length < 1);
```
nit: maybe a comment about the legacy/arcane nature of `self.attachment.filename.length < 1`

charlesmchen replied:
Done.

michaelkirk commented on Signal/src/UIColor+OWS.m:
```diff
@@ -125,4 +126,17 @@ + (UIColor *)colorWithRGBHex:(unsigned long)value
     return [UIColor colorWithRed:red green:green blue:blue alpha:1.f];
 }
 
+- (UIColor *)blendWithColor:(UIColor *)otherColor alpha:(CGFloat)alpha
```
nice.

michaelkirk commented on Signal/src/views/AudioProgressView.swift:
```diff
@@ -0,0 +1,97 @@
+        }
+    }
+
+    @available(*, unavailable, message:"use delegate: constructor instead.")
```
nit: unavailable comment seems like it should be like "use init() constructor instead"

charlesmchen replied:
Done.

charlesmchen commented:
> progress bar of the playing message bounces all around.

Fixed. PTAL @michaelkirk

michaelkirk commented on Signal/src/UIColor+OWS.m:
```diff
@@ -129,14 +129,22 @@ + (UIColor *)colorWithRGBHex:(unsigned long)value
+    result =
+#endif
+        [otherColor getRed:&r1 green:&g1 blue:&b1 alpha:&a1];
+    OWSAssert(result) return [UIColor colorWithRed:CGFloatLerp(r0, r1, alpha)
```
nit: weird spacing, missing `;` in your assert macros?

Also is the ifdef there to avoid an unused error?

charlesmchen replied:
Exactly.

charlesmchen replied:
Add I added the semicolons.

michaelkirk commented:
Sorry, I actually meant to LGTM with the last review.

But I'm curious...

> progress bar of the playing message bounces all around.

How did you fix this? I don't see any relevant changes since the last review.

charlesmchen commented:
> How did you fix this? I don't see any relevant changes since the last review.

We should cancel any audio attachment playback when we start recording a voice message.

```
   // Cancel any ongoing audio playback.
    [self.audioAttachmentPlayer stop];
    self.audioAttachmentPlayer = nil;
```
